src/test/java/io/supertokens/test/GeneratePasswordResetTokenAPITest2_4.java

```diff
@@ -43,4 +48,116 @@ public void beforeEach() {
+                io.supertokens.test.httpRequest.HttpRequest
+                        .sendJsonPOSTRequest(process.getProcess(), "",
+                                "http://localhost:3567/recipe/user/password/reset/token", requestBody, 1000,
+                                1000,
+                                null, Utils.getCdiVersion2_4ForTests());
+                throw new Exception("Should not come here");
+            } catch (io.supertokens.test.httpRequest.HttpResponseException e) {
+                assertTrue(e.statusCode == 400 &&
+                        e.getMessage()
+                                .equals("Http error. Status Code: 400. Message: Field name 'userId' is invalid in " +
+                                        "JSON input"));
+
+            }
+        }
+
+        {
+            JsonObject requestBody = new JsonObject();
+            requestBody.add("userId", null);
+            try {
+                io.supertokens.test.httpRequest.HttpRequest
+                        .sendJsonPOSTRequest(process.getProcess(), "",
+                                "http://localhost:3567/recipe/user/password/reset/token", requestBody, 1000,
+                                1000,
+                                null, Utils.getCdiVersion2_4ForTests());
+                throw new Exception("Should not come here");
+            } catch (io.supertokens.test.httpRequest.HttpResponseException e) {
+                assertTrue(e.statusCode == 400 &&
+                        e.getMessage()
+                                .equals("Http error. Status Code: 400. Message: Field name 'userId' is invalid in " +
+                                        "JSON input"));
+            }
+        }
+
+        process.kill();
+        assertNotNull(process.checkOrWaitForEvent(ProcessState.PROCESS_STATE.STOPPED));
+    }
+
+    // Check good input works
+    @Test
+    public void testGoodInput() throws Exception {
+        String[] args = {"../"};
+
+        TestingProcessManager.TestingProcess process = TestingProcessManager.start(args);
+        assertNotNull(process.checkOrWaitForEvent(ProcessState.PROCESS_STATE.STARTED));
+        
+        JsonObject signUpResponse = Utils.signUpRequest(process, "random@gmail.com", "validPass123");
+        assertEquals(signUpResponse.get("status").getAsString(), "OK");
+        String userId = signUpResponse.getAsJsonObject("user").get("id").getAsString();
+
+        JsonObject requestBody = new JsonObject();
+        requestBody.addProperty("userId", userId);
+
+        JsonObject response = io.supertokens.test.httpRequest.HttpRequest
+                .sendJsonPOSTRequest(process.getProcess(), "",
+                        "http://localhost:3567/recipe/user/password/reset/token", requestBody, 1000,
+                        1000,
+                        null, Utils.getCdiVersion2_4ForTests());
+
+        assertEquals(response.get("status").getAsString(), "OK");
+        assertNotNull(response.get("token"));
```

Comment:
Also check that the number of fields is two here (likewise for all other tests that check for good input).